test/PspUpf.jl

```diff
@@ -0,0 +1,128 @@
+            for q in (0.01, 0.1, 0.2, 0.5, 1., 2., 5., 10.)
+                reference_hgh = eval_psp_projector_fourier(hgh, i, l, q)
+                proj_upf = eval_psp_projector_fourier(upf, i, l, q)
+                @test reference_hgh ≈ proj_upf atol=1e-7 rtol=1e-7
+            end
+            for r in [upf.rgrid[1], upf.rgrid[ircut]]
+                reference_hgh = eval_psp_projector_real(hgh, i, l, r)
+                proj_upf = eval_psp_projector_real(upf, i, l, r)
+                @test reference_hgh ≈ proj_upf atol=1e-7 rtol=1e-7
+            end
+        end
+    end
+end
+
+@testset "Energy correction is consistent with HGH" begin
+    for pspfile in hgh_upf_files
+        upf = parse_upf_file("psp/$(pspfile).upf")
+        hgh = load_psp("hgh/pbe/$(pspfile).hgh")
+        n_electrons = 3
+        reference_hgh = eval_psp_energy_correction(hgh, n_electrons)
+        @test reference_hgh ≈ eval_psp_energy_correction(upf, n_electrons) rtol=1e-5 atol=1e-5
+    end
+end
+
+@testset "Potentials are consistent in real and Fourier space" begin
+    function integrand(psp, q, r)
+        4π * (eval_psp_local_real(psp, r) + psp.Zion / r) * sin(q * r) / (q * r) * r^2
+    end
+    for pspfile in all_upf_files
+        psp = parse_upf_file("psp/$(pspfile).upf")
+        for q in (0.01, 0.1, 0.2, 0.5, 1., 2., 5., 10.)
+            reference = quadgk(r -> integrand(psp, q, r), psp.rgrid[begin], psp.rgrid[end])[1]
+            correction = 4π * psp.Zion / q^2
+            @test (reference - correction) ≈ eval_psp_local_fourier(psp, q) rtol=1. atol=1.
+        end
+    end
+end
+
+@testset "Projectors are consistent in real and Fourier space" begin
+    # The integrand for performing the spherical Hankel transfrom,
+    # i.e. compute the radial part of the projector in Fourier space
+    function integrand(psp, i, l, q, x)
+        4π * x^2 * eval_psp_projector_real(psp, i, l, x) * sphericalbesselj(l, q*x)
+    end
+
+    for pspfile in all_upf_files
+        psp = parse_upf_file("psp/$(pspfile).upf")
+        ir_start = iszero(psp.rgrid[1]) ? 2 : 1 
+        for l in 0:psp.lmax, i in count_n_proj_radial(psp, l)
+            ir_cut = length(psp.projs[l+1][i])
+            for q in (0.01, 0.1, 0.2, 0.5, 1., 2., 5., 10.)
+                reference = quadgk(r -> integrand(psp, i, l, q, r),
+                                   psp.rgrid[ir_start], psp.rgrid[ir_cut])[1]
+                @test reference ≈ eval_psp_projector_fourier(psp, i, l, q) atol=1e-2 rtol=1e-2
+            end
+        end
+    end
+end
+
+@testset "PSP energy correction is consistent with fourier-space potential" begin
+    q_small = 1e-3    # We are interested in q→0 term
+    for pspfile in all_upf_files
+        psp = parse_upf_file("psp/$(pspfile).upf")
+        coulomb = -4π * (psp.Zion) / q_small^2
+        reference = eval_psp_local_fourier(psp, q_small) - coulomb
+        @test reference ≈ eval_psp_energy_correction(psp, 1) atol=1e-3
+    end
+end
```

Comment:
Why are you committing PSP8 files? Also the UPF files are quite large, would it be possible to avoid committing them, and maybe get them from the internet when running the tests? Or have it in a separate package and add it as a test dep? I'd say let's merge and do it later, but once a binary is in the git repo, it stays there forever...

Reply:
Yeah, that's probably true. The Julia ecosystem has artefacts for exactly this reason. We should look into them. For psp data they are probably very useful.

I suggest to do this after the merge. To avoid committing theme we could also just throw the files into a separate git repo for now and download them in the tests using Julias download mechanism (or directly fetch them from the net).

Reply:
Oops, didn't mean to have them in. I've removed them.